docs/rag/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,438 @@
+# Design Proposal - Embedding Ingestion Pipeline And RAG-Based Chat
+
+Not addressed in this document:
+
+* Vector store authentication options.
+* Document versioning and data update policies.
+* Unify prompt management in InstructLab. See (`chat_template` [configuration][chat_template] and
+`augment_chat_template` [function][augment_chat_template])
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+**Options to Rebuild Excalidraw Diagrams**
+
+* Using this [shareable link][shareable-excalidraw]
+* Importing the scene from the exported [DSL](./images/rag-ingestion-and-chat.excalidraw)
+* Importing the provided images in Excalidraw
+* Importing the provided images in Excalidraw
+
+## 1. Introduction
+
+This document proposes enhancements to the `ilab` CLI to support workflows utilizing Retrieval-Augmented Generation
+(RAG) artifacts within `InstructLab`. The proposed changes introduce new commands and options for the embedding ingestion
+and RAG-based chat pipelines:
+
+* A new `ilab rag` command group, feature gated behind a `ILAB_DEV_PREVIEW` environment variable.
```

Comment:
What's the point of the feature gate?

Reply:
@cdoern and @bbrowning were concerned that if this were released as dev preview without a gate that it would set expectations that this command would continue to exist, but in reality we haven't really converged on a long term CLI for this functionality.

Reply:
Gotcha - personally I would simply prefer some kind of user alert (i.e. something like "NOTE: This is an experimental command at this time - once fully supported this warning will go away) over a env var-based feature gate - but if there was a previous convo about this I won't muck up the works, I don't feel _that_ strongly about it :smile:

Reply:
Honestly, going in and trying to implement this, it seems a lot simpler to put a warning on it. Especially since some of the new options are for existing command groups, e.g. `chat.rag.enabled`.

Reply:
I would also prefer a warning, but @cdoern and @bbrowning seemed pretty firm in their insistence on a feature gate.

Reply:
After an offline discussion yesterday, where we landed is that experimental options would cause the application to simply exit with an error message if used without the dev flag being set.